### PR TITLE
feat: [pipe-20716]: first version of pr compare page

### PR DESCRIPTION
**Pull request by cjlee01**
first version of pr compare page
image

### CODE REVIEW

File: packages/ui/src/views/repo/repo-layout/index.tsx [hunk 1]
```diff
@@ -8,10 +8,18 @@ export const RepoLayout = ({ useTranslationStore }: { useTranslationStore: () =>
   const activeTab = location.pathname.split('/').pop() || 'summary'
   const { t } = useTranslationStore()
 
+  // TODO: should have a more robust function that will determine tab since there can be nested states
+  const getFinalTab = () => {
```
Comment (praneshg239) [inline packages/ui/src/views/repo/repo-layout/index.tsx:12]: can we use `useMemo` here instead of using function?

Comment (cjlee01) [reply packages/ui/src/views/repo/repo-layout/index.tsx]: done @praneshg239

File: apps/gitness/src/pages-v2/pull-request/pull-request-compare.tsx [hunk 2]
```diff
@@ -0,0 +1,382 @@
...
+  const [prBranchCombinationExists, setPrBranchCombinationExists] = useState<number | null>(null)
+  const commitSHA = '' // TODO: when you implement commit filter will need commitSHA
+  const defaultCommitRange = compact(commitSHA?.split(/~1\.\.\.|\.\.\./g))
+  const [
+    commitRange
+    //  setCommitRange  TODO: add commit view filter dropdown to manage different commits
+  ] = useState(defaultCommitRange)
+  const targetRef = useMemo(() => selectedTargetBranch.name, [selectedTargetBranch])
+  const sourceRef = useMemo(() => selectedSourceBranch.name, [selectedSourceBranch])
+  const [cachedDiff, setCachedDiff] = useAtom(changesInfoAtom)
```
Comment (praneshg239) [inline apps/gitness/src/pages-v2/pull-request/pull-request-compare.tsx:62]: Why we are using `jotai` here? Is this expected? We have `zustand` being used already, can we adopt that here and remove `jotai` from entire application?

Comment (cjlee01) [reply apps/gitness/src/pages-v2/pull-request/pull-request-compare.tsx:62]: we wont use jotai in the future i was just migrating the compare page over. Will remove that in next pr. Just wanted to break up pr before it got too large @praneshg239

**Review by shaurya-harness — APPROVED** [review]

**Pull request merged**